src/pam_p11.c

```diff
@@ -585,7 +585,7 @@ static int key_find(pam_handle_t *pamh, int flags, const char *user,
 				count--;
 			}
 		}
-
+	next_slot:
```

Comment:
I don't really like `goto` for anything else than clean up code. My suggestion would be to either copy the logic for moving to the next slot up where the locked PIN is checked or refactor this to a `for` loop, so that you can use `continue` to skip this slot.

Reply:
Ok, I used the "goto" as a hotfix, but you are right, the final solution should be more elegant. I am preparing a patch that will solve it, but I ran into another problem - when calling PKCS11_enumerate_public_keys() on an uninitialized token we stay in an infinite loop. I'm checking if it's a libp11 problem or I'll solve it with the `if (slot->token->initialized == 0 ) continue` test.

Reply:
thank you!